packages/modules/viewer-react/src/components/iModel/IModelLoader.tsx

```diff
@@ -44,6 +45,23 @@ const IModelLoader = React.memo((viewerProps: ModelLoaderProps) => {
     backstageItems,
     loadingComponent,
   } = viewerProps;
+
+  const providers = useMemo<UiItemsProvider[]>(() => {
```

Comment:
feel like this useMemo can be optimized a little bit more;
dont forget backstageItems and uiProvers are arrays. 
the way you have it you are checking for the array is defined, which `[]` will always be
you will want to check the `array.length`

<img width="291" alt="image" src="https://github.com/iTwin/viewer/assets/11051042/adaed178-0f43-42e2-8126-399621f2ecb3">

Reply:
i would also recommend not manually defining your own id for this instance of the backstageitemsprovider, and setting an id in the test app for the provider there